src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccessTokenServiceImpl.java

```diff
@@ -48,25 +53,49 @@
     private String concurRequestAccessTokenURL;
     private String concurRefreshAccessTokenURL;
     private String concurRevokeAccessTokenURL;
-    protected WebServiceCredentialService webServiceCredentialService;
+    protected AwsSecretService awsSecretService;
     protected ConcurBatchUtilityService concurBatchUtilityService;
 
+    protected ConcurStaticConfig getStaticConfig() {
+        return getConcurPojoFromAws(ConcurAwsKeyNames.STATIC_CONFIG, ConcurStaticConfig.class);
+    }
+
+    protected ConcurTokenConfig getTokenConfig() {
+        return getConcurPojoFromAws(ConcurAwsKeyNames.TOKEN_CONFIG, ConcurTokenConfig.class);
+    }
+
+    protected <T> T getConcurPojoFromAws(String awsKeyName, Class<T> pojoClass) {
+        try {
+            return awsSecretService.getPojoFromAwsSecret(awsKeyName, true, pojoClass);
```

Comment:
I'm wondering if we should go with a hybrid approach for the shared-vs-instance usage of the new DTOs. The static-config DTO could be the shared one that is used by all KFS environments. However, if we set up the token-config DTO to be instance-specific, that would allow each non-PROD environment (and our local development environments) to independently set up and manage their own access tokens. This could simplify testing the Concur Event Notification process in non-PROD environments, and will also avoid the risk of accidentally having a non-PROD environment change the PROD token.

Please let me know your thoughts on whether a shared-static-config and instance-token-config setup makes sense.

Reply:
I like the idea of being able to test in dev and test independently from prod.  I think we would need to cooridinate with Tami to set up the accounts